src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -956,6 +956,10 @@ public function setNotModified()
         $this->setStatusCode(304);
         $this->setContent(null);
 
+        if (property_exists($this, 'streamed')) {
+            $this->streamed = true;
+        }
+
```

Comment:
I don't really think this change is a good one. I've used a streamed response recently for the first time but I haven't encountered the issue you've mentioned. 

Isn't it easier to just add `setNotModified` in `StreamedResponse`, set the property to true and then call `parent::setNotModified()`?